library/common/jni/BUILD

```diff
@@ -1,13 +1,24 @@
 load("//bazel:kotlin_lib.bzl", "envoy_mobile_so_to_jni_lib")
 load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library")
-load("@envoy//bazel:envoy_build_system.bzl", "envoy_cc_library", "envoy_package")
+load("@envoy//bazel:envoy_build_system.bzl", "envoy_package")
 load("//bazel:envoy_mobile_test_extensions.bzl", "TEST_EXTENSIONS")
 
 licenses(["notice"])  # Apache 2
 
 envoy_package()
 
-envoy_cc_library(
+cc_library(
+    name = "jni_import_lib",
+    srcs = [
+        "jni_import.h"
+    ]
+    deps = select({
+        "//bazel:android_os": [],
+        "//conditions:default": [ "//bazel:jni"],
```

Comment:
might want to restrict the visibility of this so no one accidentally uses it

Reply:
How would I set the visibility so that just this file can import jni.h?

Reply:
the best you can do is the package level so `"visibility" = ["//library/common/jni:__pkg__"]` which maybe isn't restrictive enough to be worth it

Reply:
We could have a quick format check? Should be easy?

Reply:
@keith Would there be an easy way to bazel query to block only allow jni_import to have a direct dependency?

Reply:
Maybe instead of doing this you can move this rule somewhere else that would be in a more limited BUILD file so the visibility would be enough?

Reply:
> Maybe instead of doing this you can move this rule somewhere else that would be in a more limited BUILD file so the visibility would be enough?

That wouldn't prevent someone from depending on //bazel:jni though, right?

Reply:
We could put jni_import.h in its own package, then limit bazel:jni to only be visible to that package. jni_import would have public visibility so others could depend on that, but not jni.h directly

Reply:
This wouldn't prevent someone from writing another target in bazel/BUILD that pulls in jni.h, but it seems fairly robust beyond that

Reply:
> This wouldn't prevent someone from writing another target in bazel/BUILD that pulls in jni.h, but it seems fairly robust beyond that

That's why I was suggesting a format check via buildifier, deps query or something else. Seems not that hard.